content/en/docs/pipelines/sdk/python-function-components.ipynb

```diff
@@ -252,66 +374,93 @@
-    "[dockerfile]: https://docs.docker.com/engine/reference/builder/\n",
+    "The following example accepts a file as an input and returns two files as outputs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def split_text_lines(\n",
+    "    source_path: InputPath(str),\n",
+    "    odd_lines_path: OutputPath(str),\n",
+    "    even_lines_path: OutputPath(str)):\n",
+    "    \"\"\"Splits a text file into two files, with even lines going to one file\n",
+    "    and odd lines to the other.\"\"\"\n",
+    "    \n",
+    "    with open(source_path, 'r') as reader:\n",
+    "        with open(odd_lines_path, 'w') as odd_writer:\n",
+    "            with open(even_lines_path, 'w') as even_writer:\n",
+    "                while True:\n",
+    "                    line = reader.readline()\n",
+    "                    if line == \"\":\n",
+    "                        break\n",
+    "                    odd_writer.write(line)\n",
+    "                    line = reader.readline()\n",
+    "                    if line == \"\":\n",
+    "                        break\n",
+    "                    even_writer.write(line)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, the inputs and outputs are defined as parameters of the\n",
+    "`split_text_lines` function. This lets Kubeflow Pipelines pass the path to the\n",
+    "source data file and the paths to the output data files into the function.\n",
+    "\n",
+    "To accept a file as an input parameter, use one of the following type annotations:\n",
+    "\n",
+    "*   [kfp.components.InputBinaryFile][input-binary]: Use this annotation to\n",
+    "    specify that your function expects a parameter to be an\n",
+    "    [`io.BytesIO`][bytesio] instance that this function can read.\n",
+    "*   [kfp.components.InputPath][input-path]: Use this annotation to specify that\n",
+    "    your function expects a parameter to be the path to the input file as\n",
+    "    a `string`.\n",
+    "*   [kfp.components.InputTextFile][input-text]: Use this annotation to specify\n",
+    "    that your function expects a parameter to be an\n",
+    "    [`io.TextIOWrapper`][textiowrapper] instance that this function can read.\n",
+    "\n",
+    "To return a file as an output, use one of the following type annotations:\n",
+    "\n",
+    "*   [kfp.components.OutputBinaryFile][output-binary]: Use this annotation to\n",
+    "    specify that your function expects a parameter to be an\n",
+    "    [`io.BytesIO`][bytesio] instance that this function can write to.\n",
+    "*   [kfp.components.OutputPath][output-path]: Use this annotation to specify that\n",
+    "    your function expects a parameter to be the path to store the output file at\n",
+    "    as a `string`.\n",
+    "*   [kfp.components.OutputTextFile][output-text]: Use this annotation to specify\n",
```

Comment:
Thanks for the awesome update @joeliedtke 👍 

I just have one quick question - in the _Passing parameters by file_ section, do you think we should use code font for [classes](https://kubeflow-pipelines.readthedocs.io/en/latest/_modules/kfp/components/_python_op.html#InputBinaryFile) (under `kfp.components._python_op`)? For example:

```diff
- *   [kfp.components.InputBinaryFile][input-binary]
+ *   [`kfp.components.InputBinaryFile`][input-binary]
```

Thanks for all the help!!

Reply:
Great point, thanks! Done.